### PR TITLE
feat: User Testing infrastructure

### PR DESCRIPTION
**Pull request by mcdurdin**
This adds infrastructure to support user testing in keymanapp/keyman pull requests and issues:

1. Parsing of user testing specifications and test runs
2. Unit tests for user test spec/run parser
3. An implementation for @keymanapp-test-bot which watches pull requests and issues and manages reporting results.
2. Reporting on user test results
3. Status check on user test
4. Labelling user-test-missing and user-test-required for issues and pull requests
5. Refactor of status.keyman.com routes to support additional pages for future editor of test results

See https://github.com/keymanapp/keyman/wiki/User-Testing-Workflows for documentation on usage.

### CODE REVIEW

File: server/keymanapp-test-bot/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+-   a) consider using emoji for SUITES and GROUPS so we don't confuse with checkboxes ✅ 🟧 🟥 🟩 🟦 🟨 ❎ ❌
+-3. auto PRs will not be tagged by the bot. (either auto label or created by [any] bot)
+-2. icon and metadata for the bot - which orgs can install it, etc.
+-1. Deploy manual-test-bot - PEM, secret, and address.
+-5. Update wiki page for groups and suites
+
+# TO DO
+
+
+6. Edumacation and links from our internal documentation
```
Comment (rc-swag) [inline server/keymanapp-test-bot/README.md:24]: Edumacatoin -- sarcasm?

File: server/keymanapp-test-bot/keymanapp-test-bot.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+};
+
+async function processEvent(
+  octokit: InstanceType<typeof ProbotOctokit>,
+  data: ProcessEventData,
+  is_pull_request: boolean
+) {
+
+  // TODO: we may just be able to use the issue data coming in
+  const issue = await octokit.rest.issues.get({...data});
```
Comment (rc-swag) [inline server/keymanapp-test-bot/keymanapp-test-bot.ts:34]: So are issue/pr the same as far as this parser is concerned. differentiated by certain defines

**Review by rc-swag — APPROVED** [review]
lgtm

File: server/keymanapp-test-bot/keymanapp-test-bot.ts (hunk 2 above)
Comment (mcdurdin) [reply server/keymanapp-test-bot/keymanapp-test-bot.ts:34]: Yep, pull requests are essentially issue+code in GitHub, which is very convenient!

**Pull request merged**